lib/puppet/pops/lookup/hiera_config.rb

```diff
@@ -106,7 +106,18 @@ def self.config_exist?(config_root)
   def self.symkeys_to_string(struct)
     case(struct)
     when Hash
-      Hash[struct.map { |k,v| [k.to_s, symkeys_to_string(v)] }]
+      map = struct.map { |k,v|
+        case k.class.to_s
+        when "Symbol"
+          key = k.to_s
+        else
+          key = k
+        end
+        value = symkeys_to_string(v)
+        [key, value]
+      }
+
+      Hash[map]
```

Comment:
Actually, I think this might be slightly faster still:
```
map = {}
struct.each_pair {|k,v| map[ k.is_a?(Symbol) ? k.to_s : k] = symkeys_to_string(v) }
map
```
Note the use of `#each_pair` instead of `#each`. This saves Ruby from creating and splashing an array in each iteration.